components/raftstore/src/store/peer_storage.rs

```diff
@@ -1164,64 +1163,50 @@ where
     }
 
     pub fn compact_to(&mut self, idx: u64) {
-        if let Some(ref mut cache) = self.cache {
-            cache.compact_to(idx);
-        } else {
-            let rid = self.get_region_id();
-            self.engines.raft.gc_entry_cache(rid, idx);
-        }
+        self.compact_cache_to(idx);
 
         self.cancel_generating_snap(Some(idx));
     }
 
     pub fn compact_cache_to(&mut self, idx: u64) {
-        if let Some(ref mut cache) = self.cache {
-            cache.compact_to(idx);
-        } else {
-            let rid = self.get_region_id();
+        self.cache.compact_to(idx);
+        let rid = self.get_region_id();
+        if self.engines.raft.has_builtin_entry_cache() {
```

Comment:
Can it be removed completely?

Reply:
I don't know the future plan for raft engine. /cc @hicqu @tabokie

Reply:
Entry cache will be removed from RaftEngine. However for now let's keep the code.